### PR TITLE
Improve contributing docs for new developers

### PR DESCRIPTION
**Pull request by tlynam**

### CODE REVIEW

**Comment by tlynam** [comment]
Hi @zenspider, do these changes look good?

**Comment by indirect** [comment]
I test my changes directly out of my checked out repository by running `ruby -I/path/to/rubygems/lib my_script.rb`. That seems like a lot less work to me. :)

File: CONTRIBUTING.rdoc [hunk 1]
```diff
@@ -23,13 +23,23 @@ here: http://guides.rubygems.org/contributing/
...
 
-Run:
+==== To Run Tests:
 
     $ gem install hoe
     $ rake newb
 
 After `rake newb` finishes you can run `rake` to run the tests.
 
+==== To Edit Source:
```
Comment (mpapis) [inline CONTRIBUTING.rdoc]: How about: `To test manually`?

File: CONTRIBUTING.rdoc [hunk 2]
```diff
@@ -23,13 +23,23 @@ here: http://guides.rubygems.org/contributing/
 
 == Getting Started
 
-Run:
+==== To Run Tests:
```
Comment (drbrain) [inline CONTRIBUTING.rdoc]: I think this should be `=== Setup the Development Environment`

As a side-effect it happens to run tests, but you don't need to run `rake newb` to run the tests when you make changes.

**Comment by drbrain** [comment]
I agree with @indirect, using `ruby -Ilib` and `ruby -Ilib bin/gem` are the best ways to test out changes.  Installing a new ruby implies that it is very difficult to test changes to RubyGems.

**Comment by mpapis** [comment]
oh I did not thought that `-Ilib` would work as ruby requires rubygems quite early, good to know there are better ways :)

**Comment by drbrain** [comment]
`-I` is handled before ruby starts setting up a VM so it is OK.

**Comment by tlynam** [comment]
Thanks a lot for your help.  What about the setup.rb file?  Is it deprecated?

**Comment by indirect** [comment]
Thanks!

**Pull request merged**